libs/designer/src/lib/ui/panel/panelTabs/monitoringTab/monitoringTab.tsx

```diff
@@ -5,21 +5,32 @@ import { useNodeMetadata } from '../../../../core/state/workflow/workflowSelecto
 import { InputsPanel } from './inputsPanel';
 import { OutputsPanel } from './outputsPanel';
 import { PropertiesPanel } from './propertiesPanel';
+import { RunService } from '@microsoft/designer-client-services-logic-apps';
 import { ErrorSection } from '@microsoft/designer-ui';
 import type { PanelTab } from '@microsoft/designer-ui';
 import { isNullOrUndefined } from '@microsoft/utils-logic-apps';
+import { useEffect, useState } from 'react';
 
 export const MonitoringPanel: React.FC = () => {
   const selectedNodeId = useSelectedNodeId();
   const nodeMetadata = useNodeMetadata(selectedNodeId);
   const brandColor = useBrandColor(selectedNodeId);
   const runMetaData = nodeMetadata?.runData;
+  const [inputOutputs, setInputsOutputs] = useState({ inputs: {}, outputs: {} });
+
+  useEffect(() => {
+    async function getActionInputsOutputs() {
+      const actionsInputsOutputs = await RunService().getActionLinks(runMetaData, selectedNodeId);
+      setInputsOutputs(actionsInputsOutputs);
+    }
+    getActionInputsOutputs();
+  }, [runMetaData]);
```

Comment:
This might be better stored as a useQuery if we're not doing any modification of these values, then we will also get access to things like error state data and loading states

Reply:
Sure thing, I will move them to useQuery